sql/core/src/main/scala/org/apache/spark/sql/execution/basicOperators.scala

```diff
@@ -263,6 +265,10 @@ case class Except(left: SparkPlan, right: SparkPlan) extends BinaryNode {
   protected override def doExecute(): RDD[InternalRow] = {
     left.execute().map(_.copy()).subtract(right.execute().map(_.copy()))
   }
+
+  override def outputsUnsafeRows: Boolean = children.forall(_.outputsUnsafeRows)
+  override def canProcessUnsafeRows: Boolean = true
```

Comment:
Thank you! @davies @cloud-fan I added two extra test cases for ensuring they have the same formats.